src/OpenTelemetry.Exporter.Geneva/Metrics/GenevaMetricExporter.cs

```diff
@@ -500,6 +880,99 @@ private void SerializeHistogramBucket(in HistogramBucket bucket, ref int bufferI
         MetricSerializer.SerializeUInt32(this.bufferForHistogramMetrics, ref bufferIndex, Convert.ToUInt32(bucket.BucketCount));
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private void SerializeHistogramBucketWithTLV(in HistogramBucket bucket, ref int bufferIndex, double lastExplicitBound)
+    {
+        if (bucket.ExplicitBound != double.PositiveInfinity)
+        {
+            MetricSerializer.SerializeUInt64(this.buffer, ref bufferIndex, Convert.ToUInt64(bucket.ExplicitBound));
+        }
+        else
+        {
+            // The bucket to catch the overflows is one greater than the last bound provided
+            MetricSerializer.SerializeUInt64(this.buffer, ref bufferIndex, Convert.ToUInt64(lastExplicitBound + 1));
+        }
+
+        MetricSerializer.SerializeUInt32(this.buffer, ref bufferIndex, Convert.ToUInt32(bucket.BucketCount));
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private void SerializeExemplar(Exemplar exemplar, ref int bufferIndex)
+    {
+        MetricSerializer.SerializeByte(this.buffer, ref bufferIndex, 0); // version
```

Comment:
this is separate version? We write a version just before calling SerializeExemplar

Reply:
Yeah we have to serialize the same version multiple times. Once for the exemplar list, and then one for each individual exemplar. The version value is going to be zero for now. It might change in future.